File: app/src/main/java/ru/nak/ied/qrprog/data/dao/user/IUserDao.java
package ru.nak.ied.qrprog.data.dao.user;

import java.util.List;

import ru.nak.ied.qrprog.data.entities.User;

public interface IUserDao {

    public void addUser(User user);

    public User getUserById(int id);

    public User getUserByPersonNum(int personNum);

    public List<User> getAllUsers();
}

File: app/src/main/java/ru/nak/ied/qrprog/data/dao/user/IUserDao.kt
package ru.nak.ied.qrprog.data.dao.user

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import ru.nak.ied.qrprog.data.entities.User
import kotlinx.coroutines.flow.Flow

@Dao
interface IUserDao {

    @Insert
    fun insertItem(user: User)

    @Query("SELECT * FROM users")
    fun getAllUsers(): Flow<List<User>>
}